From 218e199f1d5a7bf40eb252e7e9a4c2c23b996f6c Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <Guoqing.Jiang@windriver.com>
Date: Fri, 9 Nov 2012 11:07:20 +0800
Subject: [PATCH] drivers: vexpress: add drivers infrastructure

Commit 195d7d059481310ddc8fff7076f584922bc0eb66 from
git://git.linaro.org/kernel/linux-linaro-tracking.git

This patch provides the new kernel infrastructure needed by versatile
express boards. Drivers are moved to drivers/misc/vexpress directory,
where Kconfig and Makefile are created. A coalesced include file is
created to cater for all versatile express required function declarations.

Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
[with context adjust]
Signed-off-by: Guoqing Jiang <Guoqing.Jiang@windriver.com>

diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index 61b18f8..7230792 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -546,4 +546,5 @@ source "drivers/misc/ti-st/Kconfig"
 source "drivers/misc/lis3lv02d/Kconfig"
 source "drivers/misc/carma/Kconfig"
 source "drivers/misc/altera-stapl/Kconfig"
+source "drivers/misc/vexpress/Kconfig"
 endmenu
diff --git a/drivers/misc/Makefile b/drivers/misc/Makefile
index ef4f120..77a7497 100644
--- a/drivers/misc/Makefile
+++ b/drivers/misc/Makefile
@@ -50,4 +50,5 @@ obj-$(CONFIG_USB_SWITCH_FSA9480) += fsa9480.o
 obj-$(CONFIG_ALTERA_STAPL)	+=altera-stapl/
 obj-$(CONFIG_MAX8997_MUIC)	+= max8997-muic.o
 obj-$(CONFIG_ARM_CCI)		+= arm-cci.o
+obj-$(CONFIG_ARCH_VEXPRESS)	+= vexpress/
 obj-$(CONFIG_HWLAT_DETECTOR)	+= hwlat_detector.o
diff --git a/include/linux/vexpress.h b/include/linux/vexpress.h
new file mode 100644
index 0000000..def9f78
--- /dev/null
+++ b/include/linux/vexpress.h
@@ -0,0 +1,9 @@
+/*
+ * Copyright (C) 2012 ARM Limited
+ *
+ * This file is subject to the terms and conditions of the GNU General Public
+ * License.  See the file COPYING in the main directory of this archive
+ * for more details.
+ *
+ * Versatile Express register definitions and exported functions
+ */
-- 
1.8.3.1

